src/leafletwidget/widget.ts

```diff
@@ -80,28 +77,51 @@ class MapWidget extends Widget {
    */
   protected onUpdateRequest(msg: Message): void {
     this.title.text = this._context.path.split('/').pop();
-    let cm = this._context.contentsModel;
-    if (cm === null) {
+    if (!this.isAttached) {
       return;
     }
-    let content = JSON.parse(this._context.model.toString());
+    let contentString = this._context.model.toString();
+    if (contentString === this._geojsonString) {
```

Comment:
Since I'm just comparing strings that came straight from the server side, it should be fine - I'm actually comparing file contents. At no point was it serialized to json and back again on the client side.

That said, if we provide a json editor at some point in the future, then the contents could change like you mention, so it makes more sense to compare as json objects.